src/main/java/org/opensearch/neuralsearch/plugin/query/NeuralQueryBuilder.java

```diff
@@ -54,10 +66,20 @@ public class NeuralQueryBuilder extends AbstractQueryBuilder<NeuralQueryBuilder>
 
     private static final int DEFAULT_K = 10;
 
+    private static MLCommonsClientAccessor ML_CLIENT;
+
+    public static void initialize(MLCommonsClientAccessor mlClient) {
+        NeuralQueryBuilder.ML_CLIENT = mlClient;
+    }
+
```

Comment:
Question: Can we init the MLCommonsClientAccessor via the constructor of NeuralQueryBuilder class? is it not possible to do so?

Reply:
Good question. I thought a little bit about this, but I am not sure about how to handle stream constructors. Any ideas on this?

Reply:
Sure, will leave as is